tests/test_block_utils.cc

```diff
@@ -107,4 +107,30 @@ TEST(test_block_utils, test_matrix_l) {
   EXPECT_LE((block_result - dense_result).norm(), 1e-6);
 }
 
+TEST(test_block_utils, test_block_symmetric) {
+
+  Eigen::Index k = 5;
```

Comment:
A lot of this setup is duplication of the struct `BlockSymmetricMatrix : public SerializableType<BlockSymmetric>`  made for the serialization test. Can it be reused here? Some sort of example/test_blocksymetric?

Reply:
Yeah I'll throw a method in `test_utils.h`